typegate/src/planner/args.ts

```diff
@@ -539,7 +549,12 @@ class ArgumentCollector {
     this.addPoliciesFrom(typeIdx);
 
     if ("injection" in typ) {
-      return this.collectInjection(typ);
+      const compute = this.collectInjection(typ);
+      if (compute != null) {
+        return compute;
+      }
+      // no injection for the effect
```

Comment:
if null is return, does the final error message depends on the type injection? why not throw directly from the collectInjection ?